.github/workflows/generate-docs.yml

```diff
@@ -0,0 +1,128 @@
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          # Fetch history for all branches and tags to allow for successful merge of base branch if needed.
+          fetch-depth: 0
+
+      - name: Determine branch names
+        id: branches
+        run: |
+          BASE_BRANCH=${GITHUB_REF#refs/heads/}
+          echo "::set-output name=base::$(echo "$BASE_BRANCH")"
+          echo "::set-output name=head::$(echo "update/docs-$BASE_BRANCH")"
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: '7.4'
+
+      - name: Get Composer Cache Directory
+        id: composer-cache
+        run: echo "::set-output name=dir::$(composer config cache-files-dir)"
+
+      - name: Configure Composer cache
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.composer-cache.outputs.dir }}
+          key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-composer-
+
+      - name: Install Composer dependencies
+        run: composer install --prefer-dist --optimize-autoloader --no-progress --no-interaction
+
+      - name: Setup WordPress
+        run: |
+          docker-compose up -d
+
+          echo "::group::Get environment vars from containers"
+          MYSQL_PWD=$(docker-compose exec -T mysql printenv MYSQL_ROOT_PASSWORD)
+          WORDPRESS_DB_NAME=$(docker-compose exec -T wordpress printenv WORDPRESS_DB_NAME)
+          HOST_PORT=$(docker-compose port wordpress 80 | awk -F : '{printf $2}')
+          echo "::endgroup::"
+
+          echo "::group::Wait for database container to be ready"
+          echo -en "Waiting for database connection..."
+          until $(docker-compose exec -T wordpress bash -c "echo -n > /dev/tcp/mysql/3306" >/dev/null 2>&1); do
+              echo -n '.'
+              sleep 5
+          done
+          echo ''
+          echo "::endgroup::"
+
+          echo "::group::Installing WordPress"
+          docker-compose exec -T -e MYSQL_PWD="$MYSQL_PWD" mysql mysql -e "CREATE DATABASE IF NOT EXISTS $WORDPRESS_DB_NAME"
+          docker-compose exec -T cli wp core install \
+            --title="Docs" --admin_user=admin --admin_password=password --admin_email=test@test.com --skip-email --url=http://localhost:$HOST_PORT
+          echo "::endgroup::"
+
+          echo "::group::Activate AMP plugin"
+          # Building the JS assets is required for the plugin to activate successfully, but since we're only generating
+          # the docs we simply create the file being checked for to save some time.
+          touch ../../assets/js/amp-block-editor.js
+          docker-compose exec -T cli wp plugin activate amp
+          echo "::endgroup::"
+        working-directory: bin/local-env
+
+      - name: Configure git user
+        run: |
+          git config user.email "pierregordon@protonmail.com"
+          git config user.name "Pierre Gordon"
```

Comment:
The `github-actions` bot isn't on the CLA allowlist so I'm using myself for now to make commits.